substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -154,26 +156,43 @@ private static long init(int level, int strategy, boolean nowrap) {
         }
     }
 
-    // Deflater.c-Java_java_util_zip_Deflater_setDictionary(JNIEnv *env, jclass cls, jlong addr,
+    // (JDK 11)src/java.base/share/native/libzip/Deflater.c-static jlong doDeflate(JNIEnv *env, jobject this, jlong addr,
+    //                       jbyte *input, jint inputLen,
+    //                       jbyte *output, jint outputLen,
+    //                       jint flush, jint params)
+    private static void doSetDictionary(long addr, CCharPointer buf, int len) {
+        int res = ZLib.deflateSetDictionary(WordFactory.pointer(addr), buf, len);
+        if (res == ZLib.Z_OK()) {
+            return;
+        } else if (res == ZLib.Z_STREAM_ERROR()) {
+            throw new IllegalArgumentException();
+        } else {
+            throw new InternalError();
```

Comment:
Is there some point in preserving the error message `((z_stream *)jlong_to_ptr(addr))->msg`?

Reply:
I know that was not done in `init` (above), but that is not an excuse to be sloppy.